pageserver/src/tenant/upload_queue.rs

```diff
@@ -81,6 +85,18 @@ impl UploadQueueInitialized {
     pub(super) fn no_pending_work(&self) -> bool {
         self.inprogress_tasks.is_empty() && self.queued_operations.is_empty()
     }
+
+    pub(super) fn get_last_remote_consistent_lsn_visible(&mut self) -> Option<Lsn> {
+        while let Ok(lsn) = self.visible_remote_consistent_lsn_rx.try_recv() {
+            self.visible_remote_consistent_lsn = Some(lsn);
+        }
+
+        self.visible_remote_consistent_lsn
+    }
```

Comment:
This seems odd for the rx and tx being in the same place, and &mut getter and ... Will the queue be filled up if no one is requesting this?

Perhaps there should be an `Arc<AtomicLsn>` or `Arc<AtomicU64>` instead?

Reply:
I did not yet find where this is written to.

Reply:
The _tx was cloned and passed into the deletion queue when requesting an update.

This is now replaced with an Arc+Atomic